src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -1217,7 +1217,9 @@ public void Dispose()
                 }
             });
 
-            ea.Execute(() => ConflictResolver?.ResolveConflictsTask.Wait());
+            ea.Execute(() => ConflictResolver?.ResolveConflictsTask.Wait(TimeSpan.FromSeconds(15)));
```

Comment:
I'm not sure about the consequences of not waiting through here, and we do a similar unbounded wait few line above, although I agree that unbounded waits in dispose function is a bad practice in general.
```suggestion
            ea.Execute(() => ConflictResolver?.ResolveConflictsTask.Wait());
```